### PR TITLE
Farm Mutable

### LINKED ISSUE #35212: Countryside seems devoid of life
**Is your feature request related to a problem? Please describe.**  
The game's countryside seems to devoid of life. Rural New England is indeed heavily forested, but IMHO the game doesn't place nearly enough farm fields - this makes the countryside less believable.

Also, there are not nearly enough farm fields, meaning that farms don't seem to be producing much. 

**Describe the solution you'd like**  
1) A higher farm field to farm ratio. There should be more farm fields in the vicinity of farms, creating clusters - a farm with a barn and a few fields nearby. 
2) Agricultural zones on map should be larger, and denser, mimicking the proximity of farms to each other, especially along rivers.
3) Country dirt roads are not as prevalent as they should be - farms should IMHO be connected to their fields using dirt roads. 

I know it's not really a priority, but a better countryside would make the world much more plausible.

### PR DESCRIPTION
**Pull request by FarkasKBence**
#### Summary
Content "Farm Mutable"

#### Purpose of change

Partially resolves #35212

New England's countryside doesn't spawn nearly enough farms, and the ones that spawn don't seem to produce much.
This PR adds a new type of farm - made with the power of mutables - that should hopefully negate the problems mentioned in the issue above.

#### Describe the solution

The mutable is made up of 9 phases:

1. **farm entrance** — the root of the mutable - connects to the farm house and the farm clusters
2. **farm house** — basically ```3x3 Farm``` and ```farm_lot_forest```
3. **farm house extension** _(if farm house is ```farm_home_1```)_ — adds either a greenhouse, silo or chicken coop to ```3x3 Farm```
4. **farm cluster** — 9x9 size farmlands that connect in a grid-like pattern
5. **farm cluster edges** _(all)_ — connects to an unresolved join of a farm cluster
6. **farm cluster edges** _(left)_ — connects to an unresolved join of a farm cluster _(fallback for 5, in case it left out a join)_
7. **farm cluster edges** _(right)_ — connects to an unresolved join of a farm cluster _(fallback for 5, in case it left out a join)_
8. **farm exit** — adds a road connection to an unresolved join of a farm cluster _(max 1)_
9. **fix unresolved joins** — fixes any remaining unresolved joins with a dummy join

A finished mutable looks something like this:

kép

The building blocks that make up the mutable mainly use the general-purpose tiles I made back in #31188, which have been expanded with newer mapgen _(like hay- and cornfields)_, and updated _(dirt roads)_ to better suit the new farm.

Some of the farms with lower crop yields ( ```farm_lot_wired_2x3```, ```farm_lot_empty_3x2```, ```farm_lot_2x4``` ) have been obsoleted. The mutable has made them redundant.

```farm_lot_forest```, ```Farm```, ```Farm with silo```, ```Farm with chicken coop``` and ```Farm with greenhouse``` are now incorporated into the mutable. They no longer spawn separately.

Other farms, like ```farm_horse```, ```St_Johns_farm```, ```farm_2side``` and ```farm_stills``` are unaffected by this PR.

#### Describe alternatives you've considered

- not obsoleting the old farms
- obsoleting the old farm fields in ```farm_tiles.json```
- adding ```farm_2side``` and ```farm_stills``` to the available building types
- differentiating silos and grain bins
- adding more variations: pastures, vegetable fields, sheds, etc...
- (I know very little about proper plowing patterns)

#### Testing

Created a new world, surveyed the land for farm mutables.

#### Additional context

A guide for all the building blocks:
kép

<details>

  <summary>More screenshots:</summary>

kép
The new dirt roads and farm fields:
kép

</details>

### CODE REVIEW

**Comment by anothersimulacrum** [comment]
This is amazing, fantastic work!

File: data/json/obsoletion/overmap_special/specials.json [hunk 1]
```diff
@@ -337,5 +337,208 @@
...
+      { "point": [ 1, 2, 0 ], "overmap": "farm_8_north" },
+      { "point": [ 2, 2, 0 ], "overmap": "farm_7_north" },
+      { "point": [ 0, 0, 1 ], "overmap": "farm_3_roof_north" },
+      { "point": [ 1, 0, 1 ], "overmap": "farm_2_roof_north" }
+    ],
+    "connections": [ { "point": [ 1, -1, 0 ], "terrain": "road", "connection": "local_road", "from": [ 1, 0, 0 ] } ],
+    "locations": [ "field" ],
+    "city_distance": [ 5, 40 ],
+    "city_sizes": [ 1, -1 ],
+    "occurrences": [ 0, 0 ],
```
Comment (Maleclypse) [inline data/json/obsoletion/overmap_special/specials.json:361]: Several of these are set to never spawn. Maybe all.  I haven't looked through all the specials yet.

Comment (Maleclypse) [inline data/json/obsoletion/overmap_special/specials.json:361]:
```suggestion
    "occurrences": [ 0, 2 ],
```
0, 2 or three would be my suggestion

**Review by Maleclypse — COMMENTED** [review]
I'm having trouble seeing what monsters would spawn in these specials.  Am I missing the spawns?

File: data/json/obsoletion/overmap_special/specials.json (hunk 1 above)
Comment (anothersimulacrum) [reply data/json/obsoletion/overmap_special/specials.json:361]: This is an obseletion file, that seems intentional.

Comment (Maleclypse) [reply data/json/obsoletion/overmap_special/specials.json:361]: Apparently work was harder on my mind than I thought today.

**Review by MNG-cataclysm — APPROVED** [review]

**Comment by LovamKicsiGazsii** [comment]
> I'm having trouble seeing what monsters would spawn in these specials. Am I missing the spawns?

Only the farmhouses and orchards have set spawns. Farm fields and dirt roads don't spawn any monsters.

**Comment by Paprika-H** [comment]
This is beautiful, thank you

**Comment by JonathanLochridge** [comment]
These look super cool. I am really looking forward to coming across these.

Obsoleting most of the old farms is cool.

I was wondering how much variation in size this caused?

I know for orchards that they tend to be smaller and more self contained more often than grains like corn and such which take massive amount of land to make sense. Since the most efficient way to farm stuff like that is with big tractors. But, orchards are often still pretty compact

**Pull request merged**